src/aci/aci_build_F.cc

```diff
@@ -864,4 +862,284 @@ void AdaptiveCI::get_core_excited_determinants(SharedMatrix evecs, DeterminantHa
+        std::vector<int> bocc = det.get_beta_occ(nmo);
+        std::vector<int> avir = det.get_alfa_vir(nmo);
+        std::vector<int> bvir = det.get_beta_vir(nmo);
+
+
+        Determinant new_det(det);
+        // Generate alpha excitations
+        for( int h = 0; h < nirrep_; ++h){
+            for (auto& ii : noalpha[h]) {
+                for (auto& aa : nvalpha[h]) {
+                    new_det = det;
+                    new_det.set_alfa_bit(ii, false);
+                    new_det.set_alfa_bit(aa, true);
+                    size_t hash_val = Determinant::Hash()(new_det);
+                    if ( (hash_val % nbin) == bin) {
+                        double HIJ = fci_ints_->slater_rules_single_alpha(det, ii, aa) * c_I;
+                        if ((std::fabs(HIJ) >= screen_thresh_)) {
+                            if (A_b.count(new_det) == 0) {
+                                A_b[new_det] = HIJ;
+                            } else {
+                                A_b[new_det] += HIJ;
+                            }
+                        }
+                    }
+                }
+            }
+            // Generate beta excitations
+            for (auto& ii : nobeta[h]) {
+                for (auto& aa : nvbeta[h]) {
+                    // Check if the determinant goes in this bin
+                    new_det = det;
+                    new_det.set_beta_bit(ii, false);
+                    new_det.set_beta_bit(aa, true);
+                    size_t hash_val = Determinant::Hash()(new_det);
+                    if ((hash_val % nbin) == bin) {
+                        double HIJ = fci_ints_->slater_rules_single_beta(det, ii, aa) * c_I;
+                        if ((std::fabs(HIJ) >= screen_thresh_)) {
+                            if (A_b.count(new_det) == 0) {
+                                A_b[new_det] = HIJ;
+                            } else {
+                                A_b[new_det] += HIJ;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        for ( int p = 0; p < nirrep_; ++p){
+            for ( int q = p; q < nirrep_; ++q){
+                for ( int r = 0; r < nirrep_; ++r){
+                    int sp = p^q^r;
+                    if( sp < r) continue;
+
+                    // Generate aa excitations
+                    for (int i = 0, max_i = noalpha[p].size(); i < max_i; ++i) {
+                        int ii = noalpha[p][i];
+                        for (int j = (p == q ? i + 1 : 0), maxj = noalpha[q].size(); j < maxj; ++j) {
+                            int jj = noalpha[q][j];
+                            for (int a = 0, max_a = nvalpha[r].size(); a < max_a; ++a) {
+                                int aa = nvalpha[r][a];
+                                for (int b = (r == sp ? a + 1 : 0), maxb = nvalpha[sp].size(); b < maxb; ++b) {
+                                    int bb = nvalpha[sp][b];
```

Comment:
Here it might be convenient to have a reference up in the loop structure `const auto& nvalpha_sp = nvalpha[sp]` so that you can call `nvalpha_sp[b]` instead. I am not sure it will be faster, but it can avoid doing a double lookup. This might be the most critical point, but the same comment applies to `ii`, `jj`, and `aa`.